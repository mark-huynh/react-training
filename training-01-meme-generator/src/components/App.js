import React from "react";

class App extends React.Component {
  state = {
    topLine: "",
    bottomLine: "",
    imageUrl: ""
  };

  componentDidMount() {
    this.setState({ message: "Hello World" });
  }

  handleLineChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  };

  render() {
    return (
      <React.Fragment>
        <label>Top Line</label>
        <input
          name="topLine"
          onChange={this.handleLineChange}
          value={this.state.topLine}
        />
        <label>Bottom Line</label>
        <input
          name="bottomLine"
          onChange={this.handleLineChange}
          value={this.state.bottomLine}
        />
        <label>Image Url</label>
        <input
          name="imageUrl"
          onChange={this.handleLineChange}
          value={this.state.imageUrl}
        />
        <h1>{this.state.topLine}</h1>
        <h1>{this.state.bottomLine}</h1>
        <img src={this.state.imageUrl} />
      </React.Fragment>
    );
  }
}

export default App;
