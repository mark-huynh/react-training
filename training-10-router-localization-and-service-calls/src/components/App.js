import React from "react";
import Home from "./Home";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { IntlProvider, addLocaleData } from "react-intl";
import en_locale_data from "react-intl/locale-data/en";
import es_locale_data from "react-intl/locale-data/es";
import en from "../i18n/en.json";
import es from "../i18n/es.json";
import Dogs from "./Dogs";

addLocaleData([...en_locale_data, ...es_locale_data]);

const messageMapping = {
  en,
  es
};

const locale = navigator.language.split("-")[0];

const App = () => {
  return (
    <IntlProvider locale={locale} messages={messageMapping[locale]}>
      <Router>
        <Route exact path="/" component={Home} />
        <Route exact path="/dogs" component={Dogs} />
      </Router>
    </IntlProvider>
  );
};

export default App;
