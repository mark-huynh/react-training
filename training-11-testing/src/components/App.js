import React, { Component } from "react";
import PropTypes from "prop-types";
import Input from "./Input";
import Meme from "./Meme";

export default class App extends Component {
  static propTypes = {
    prop: PropTypes
  };

  state = {
    topLine: "",
    bottomLine: ""
  };

  handleChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  };

  render() {
    return (
      <div>
        <Input
          handleChange={this.handleChange}
          topLine={this.state.topLine}
          bottomLine={this.state.bottomLine}
        />
        <Meme topLine={this.state.topLine} bottomLine={this.state.bottomLine} />
      </div>
    );
  }
}
