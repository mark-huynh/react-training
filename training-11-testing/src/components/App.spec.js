import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";

describe("<App />", () => {
  const INPUT_TEXT = "Look at this test, isn't this neat";
  const event = { target: { value: INPUT_TEXT } };

  describe("Snapshot Tests", () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("Unit Tests", () => {
    it("should setState to the value of the target that is passed in", () => {
      const wrapper = shallow(<App />);
      const topLineEvent = {
        ...event,
        target: { ...event.target, name: "topLine" }
      };
      wrapper.instance().handleChange(topLineEvent);
      expect(wrapper.state("topLine")).toEqual(INPUT_TEXT);
    });
  });

  describe("Integration Tests", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(<App />);
    });

    it("should display the top line of text when something is entered into the topLine input", () => {
      const topLineInput = wrapper.find("#top-line-input");
      const topLineEvent = {
        ...event,
        target: { ...event.target, name: "topLine" }
      };
      topLineInput.simulate("change", topLineEvent);
      const topLineText = wrapper.find("#top-line-text");
      expect(topLineText.text()).toEqual(INPUT_TEXT);
    });

    it("should display the bottom line of text when something is entered into the bottomLine input", () => {
      const bottomLineInput = wrapper.find("#bottom-line-input");
      const bottomLineEvent = {
        ...event,
        target: { ...event.target, name: "bottomLine" }
      };
      bottomLineInput.simulate("change", bottomLineEvent);
      const bottomLineText = wrapper.find("#bottom-line-text");
      expect(bottomLineText.text()).toEqual(INPUT_TEXT);
    });
  });
});
