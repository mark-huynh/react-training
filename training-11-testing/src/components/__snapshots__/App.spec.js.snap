exports[`undefined 1`] = `
<App>
  <div>
    <Input
      bottomLine=""
      handleChange={[Function]}
      topLine="">
      <div>
        <input
          id="top-line-input"
          name="topLine"
          onChange={[Function]}
          value="" />
        <input
          id="bottom-line-input"
          name="bottomLine"
          onChange={[Function]}
          value="" />
      </div>
    </Input>
    <Meme
      bottomLine=""
      topLine="">
      <div>
        <h1
          id="top-line-text" />
        <h1
          id="bottom-line-text" />
      </div>
    </Meme>
  </div>
</App>
`;
